nixos/modules/misc/mandoc.nix

```diff
@@ -5,59 +5,191 @@ let
 
   cfg = config.documentation.man.mandoc;
 
-in {
+  toMandocOutput = output: lib.concatLines (
+    lib.mapAttrsToList
+      (
+        name: value:
+          if lib.isString value || lib.isPath value then "output ${name} ${value}"
+          else if lib.isInt value then "output ${name} ${builtins.toString value}"
+          else if lib.isBool value then lib.optionalString value "output ${name}"
+          else throw "Unrecognized value type ${builtins.typeOf value} of key ${name} in mandoc output settings"
```

Comment:
I think you need to add a case for `null` as well, don't you? In that case no output directive should be created, right?